src/test/groovy/steps/DataAnalyticsSteps.groovy

```diff
@@ -5,25 +5,19 @@ import io.restassured.response.Response
 import org.testng.Assert
 import org.testng.annotations.DataProvider
 import org.yaml.snakeyaml.Yaml
+import tests.TestSetup
 import utils.Utils
 
-class DataAnalyticsSteps {
+class DataAnalyticsSteps extends TestSetup{
 
     def repoSteps = new RepositorySteps()
     def securitySteps = new SecuritytSteps()
     def xraySteps = new XraySteps()
     def utils = new Utils()
     def artifact = new File("./src/test/resources/repositories/artifact.zip")
     def repoListHA = new File("./src/test/resources/repositories/CreateDefault.yaml")
-    Yaml yaml = new Yaml()
-    def configFile = new File("./src/test/resources/testenv.yaml")
-    def config = yaml.load(configFile.text)
-    def protocol = config.artifactory.protocol
-    def xrayBaseUrl = "${protocol}${config.artifactory.external_ip}/xray/api"
-    def artifactoryBaseURL = "${protocol}${config.artifactory.external_ip}/artifactory"
-    def username = config.artifactory.rt_username
-    def password = config.artifactory.rt_password
-
+    def xrayURL = "${artifactoryBaseURL}/xray/api"
```

Comment:
For the record, I don't like to use `Gstring` - they aren't java strings and can have some incompatibilities. Just an alternative: `def xrayUrl = String.format('%s/xray/api',artifactoryBaseUrl)` and if you statically import `String` it just becomes `format`

Reply:
Thanks for suggesting, but "${artifactoryBaseURL}/xray/api" has better readability.